kornia/contrib/extract_patches.py

```diff
@@ -167,17 +174,28 @@ def combine_tensor_patches(
             f"Only stride == window_size is supported. Got {stride} and {window_size}."
             "Please feel free to drop a PR to Kornia Github."
         )
+
+    if original_size[0] % 2 != 0 or original_size[1] % 2 != 0:
+        raise NotImplementedError(f"Original image size must be divisible by 2. Got {original_size}")
+
     if unpadding is not None:
+        hpad_check = (original_size[0] + unpadding[0] + unpadding[1]) % window_size[0] == 0
+        wpad_check = (original_size[1] + unpadding[2] + unpadding[3]) % window_size[1] == 0
+
+        if not hpad_check or not wpad_check:
+            raise NotImplementedError("Insufficient padding")
+
         window_size = (
-            window_size[0] + (unpadding[0] + unpadding[1]) // window_size[0],
-            window_size[1] + (unpadding[2] + unpadding[3]) // window_size[1]
+            (original_size[0] + (unpadding[0] + unpadding[1])) // window_size[0],
```

Comment:
I am wondering what is the definition of `window_size` here? why it changes so dramatically?`window_size==image_size`?

Reply:
The window_size that is passed to `combine_tensor_patches` is the window_size used for `extract_tensor_patches`. 
window_size is not the same as image size.


Consider the following code snippet


```
import torch

from kornia.contrib import CombineTensorPatches, ExtractTensorPatches

h, w = 8, 8
win = 4
pad = 2

image = torch.randn(1, 1, h, w)
print(image.shape)
tiler = ExtractTensorPatches(window_size=win, stride=win, padding=pad)
merger = CombineTensorPatches(original_size=(h, w), window_size=win,  unpadding=pad)
image_tiles = tiler(image)
print(image_tiles.shape)
new_image = merger(image_tiles)
print(new_image.shape)
assert (image == new_image).all()
```


Here 
original_image_size = (8,8)
window_size used for extraction = (4, 4) 
padding = (2,2,2,2)

In combine_tensor_patches we compute window_size to take unpadding into account, so
```
combine_window_size = window_size = (
            (original_size[0] + (unpadding[0] + unpadding[1])) // window_size[0],
            (original_size[1] + (unpadding[2] + unpadding[3])) // window_size[1],
        )
```
which comes out to be (3, 3)

Reply:
would nice to have in a posterior PR this snippet as an example with visualisation

Reply:
Sure. I could do that in a follow up PR. Where do you recommend I add it?

Reply:
Great, https://github.com/kornia/tutorials/tree/master/source